### PR TITLE
Truncate strings in chat list preview

### LINKED ISSUE #4049: Add "..." to the end of chatlist's message preview if the message text doesn't fit
Reasons:
- That's already done for long chat names, so shouldn't be difficult.
- On e.g. Android "..." is added if the message text doesn't fit. Would be good to unify UIs then.

This is minor, but sometimes it looks weird:
Google Hell

### PR DESCRIPTION
**Pull request by maxphilippov**
- add truncated-class to text preview in chat list
- removed excessive div from chat message preview
- resolves #4049

### CODE REVIEW

**Comment by maxphilippov** [comment]
I've removed the excessive <div> wrapping for messages. There're also some corner cases like this on the image, but this looks like a parser refactoring to me.
Here there are multiple elements and text is split by an 'inline-code' block:
delta_issue_4049

**Comment by iequidoo** [comment]
> - resolves https://github.com/deltachat/deltachat-desktop/pull/4059

The link should be fixed in the commit message

**Comment by maxphilippov** [comment]
> > * resolves [Truncate strings in chat list preview #4059](https://github.com/deltachat/deltachat-desktop/pull/4059)
> 
> The link should be fixed in the commit message: #4049

Oh, sorry, I just looked at PR id and thought I messed up when creating my branch, turns out I was right the first time :)

**Comment by meganoahj** [comment]
Weird splitting of strings is happening sometimes, please revisit

Screenshot_20240802_230727

**Comment by maxphilippov** [comment]
> Weird splitting of strings is happening sometimes, please revisit
> 
> Screenshot_20240802_230727

Just to make sure, do you have Markdown rendering enabled in settings?

**Comment by maxphilippov** [comment]
Made it simpler, fixed issues, ready for re-review

File: src/renderer/components/message/MessageMarkdown.tsx [hunk 1]
```diff
@@ -111,23 +111,19 @@ function renderElementPreview(elm: ParsedElement, key?: number): JSX.Element {
...
-      return <s key={key}>{elm.c.map(renderElement)}</s>
+      return <s key={key}>{elm.c.map(renderElementPreview)}</s>
 
     case 'Italics':
-      return <i key={key}>{elm.c.map(renderElement)}</i>
+      return <i key={key}>{elm.c.map(renderElementPreview)}</i>
 
     case 'Bold':
-      return <b key={key}>{elm.c.map(renderElement)}</b>
+      return <b key={key}>{elm.c.map(renderElementPreview)}</b>
```
Comment (WofWca) [inline src/renderer/components/message/MessageMarkdown.tsx:120]: This fixes a separate issue, doesn't it? It's worth mentioning it in the commit and the changelog I think if so.

File: src/renderer/components/chat/ChatListItem.tsx [hunk 2]
```diff
@@ -116,7 +116,7 @@ function Message({
             }}
           />
         )}
-        <div>{message2React(summaryText2 || '', true)}</div>
+        {message2React(summaryText2 || '', true)}
```
Comment (WofWca) [inline src/renderer/components/chat/ChatListItem.tsx:119]: Was it necessary to make things work to remove the div?

**Review by WofWca — APPROVED** [review]
Apart from changelog and commit issues, code looks good. I tested it lightly - also good.

File: src/renderer/components/chat/ChatListItem.tsx (hunk 2 above)
Comment (maxphilippov) [reply src/renderer/components/chat/ChatListItem.tsx:119]: I essentially moved that inside message2React as `<div className='truncated'>`. I thought it's a part of preview logic.

File: CHANGELOG.md [hunk 3]
```diff
@@ -29,6 +29,8 @@
   - Display vCard contact name in the message summary
   - Case-insensitive search for non-ASCII messages
 - upgrade `electron` from `30.0.2` to `30.3.1`
+- truncate message previews in chat list #4059
+- renderElementPreview calls renderElementPreview for message element children #4059
```
Comment (WofWca) [inline CHANGELOG.md:33]: Too technical I'd say, but fine

**Pull request merged**